scapy/utils.py

```diff
@@ -1899,6 +1919,31 @@ def _read_block_dsb(self, block, size):
         else:
             warning("PcapNg: Unknown DSB secrets type (0x%x)!", secrets_type)
 
+    def _read_block_pib(self, block, _):
+        # type: (bytes, int) -> None
+        """Apple Process Information Block"""
+
+        # Get the Process ID
+        try:
+            dpeb_pid = struct.unpack(self.endian + "I", block[:4])[0]
+            process_information = {"id": dpeb_pid}
+            block = block[4:]
+        except struct.error:
+            warning("PcapNg: DPEB is too small (%d). Cannot get PID!",
+                    len(block))
+            raise EOFError
+
+        # Get Options
+        options = self._read_options(block)
+        for code, value in options.items():
+            if code == 2:
+                process_information["name"] = value.decode("ascii", "backslashreplace")
+            elif code == 4:
+                process_information["uuid"] = str(UUID(bytes=value))
```

Comment:
I tested it with https://github.com/secdev/scapy/pull/4378 and the fuzz target crashed eventually with
```
ValueError: bytes is not a 16-char string
Traceback (most recent call last):
  File "/home/vagrant/scapy/pcap_fuzzer.py", line 30, in TestOneInput
    scapy.utils.rdpcap(io.BytesIO(input_bytes))
  File "/home/vagrant/scapy/scapy/utils.py", line 1261, in rdpcap
    return fdesc.read_all(count=count)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1538, in read_all
    res = self._read_all(count)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1448, in _read_all
    p = self.read_packet()  # type: Packet
        ^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1962, in read_packet
    rp = super(PcapNgReader, self)._read_packet(size=size)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1693, in _read_packet
    res = self._read_block()
          ^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1620, in _read_block
    return self.blocktypes[blocktype](block, size)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1942, in _read_block_pib
    process_information["uuid"] = str(UUID(bytes=value))
                                      ^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/uuid.py", line 187, in __init__
    raise ValueError('bytes is not a 16-char string')
ValueError: bytes is not a 16-char string

==28164== ERROR: libFuzzer: fuzz target exited
```

Reply:
The hexdump of its input was
```
00000000  0a 0d 0d 0a 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000010  00 14 00 00 00 4b 53 00  01 00 08 ab 20 00 00 00  |.....KS..... ...|
00000020  01 00 00 80 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000030  01 00 00 00 00 01 00 00  01 00 08 ab 20 00 00 00  |............ ...|
00000040  01 00 00 80 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000050  04 00 00 00 01 00 00 00  1a 01 00 af 20 00 00 00  |............ ...|
00000060
```